content/docs/rfcs/19/README.md

```diff
@@ -0,0 +1,101 @@
+---
+slug: 19
+title: 19/WAKU2-DIRECT-MESSAGE
+name: Waku v2 Direct Message
+status: raw
+editor: Franck Royer <franck@status.im>
+contributors:
+---
+
+This specification explains the Waku v2 Direct Message protocol which enables a peer to send a direct message to another peer using the Waku v2 network.
+
+# Send message only knowing recipient's Ethereum Public Key
+
+This section describes how to use the Waku v2 Direct Message protocol to send a message to another party using their Ethereum Public Key.
+In this section, we assume that Alice sends a direct message to Bob.
+
+
+## Variables
+
+Here are the variables used in the protocol and their definition:
+
+- `A` is Alice's Ethereum root HD public key,
+- `B` is Bob's Ethereum root HD public key,
+- `a` is the private key of `A`, and is only known by Alice,
+- `b` is the private key of `B`, and is only known by Bob.
+
+## Design Requirements
+
+The proposed protocol MUST adhere to the following design requirements:
+
+a. Alice knows Bob's Ethereum root HD public key `B`,
+b. Alice wants to send message `M` to Bob using waku,
+c. Bob SHOULD be able to get `M` using [13/WAKU2-STORE](/spec/13), when querying a store node that hosts `M`,
+d. Bob MUST recognize he is `M`'s recipient when relaying it via [11/WAKU2-RELAY](/spec/11),
+e. Carole MUST NOT be able to read `M`'s content even if she is storing it or relaying it.
+
+## Steps
+
+1. Alice MUST derive Bob's waku public Key `Bw` from `B`,
```

Comment:
My understanding is that it is hard to compute `B` from `Bw`. Which means that just by observing the waku traffic, one cannot deduce which Ethereum addresses are receiving message.

However, we know that as highlighted in https://github.com/vacp2p/rfc/pull/365#discussion_r626160046 if one knows `B`, it is easy to know whether Bob is receiving messages.

I also expect that as we evolve the protocol, we use new ways to derive keys from `B` to allow unlinkability.

What do you think?

Reply:
If the derivation is based on a deterministic function F i.e., F always output the same Bw when receiving B, then anonymity is still at risk. This is regardless of the derivation function F being one-way (one-way: computing B from Bw  is hard). 
The anonymity is compromised since all the network messages in the network are linkable to their receiver  by only looking at the embedded pk i.e., Bw.